core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -818,13 +818,12 @@ class DAGScheduler(
       }
     }
 
-    val properties = if (jobIdToActiveJob.contains(jobId)) {
-      jobIdToActiveJob(stage.jobId).properties
-    } else {
-      // this stage will be assigned to "default" pool
-      null
+    val activeJob = jobIdToActiveJob.get(stage.jobId).getOrElse(null)
```

Comment:
Are you suggesting that in some cases `jobId != stage.jobId` and that's the error?
Then just `val properties = jobIdToActiveJob.get(stage.jobId).map(_.properties).getOrElse(null)`?